src/corecel/data/Copier.cc

```diff
@@ -62,13 +63,14 @@ void copy_bytes(MemSpace dstmem,
                 MemSpace srcmem,
                 void const* src,
                 std::size_t count,
-                CELER_UNUSED_UNLESS_DEVICE StreamId stream)
+                StreamId stream)
```

Comment:
In the case where the variable is actually never used in any configuration, should we keep the original (i.e. `[[maybe_unused]]`)?

Reply:
The problem is that GCC seems to warn when it's "never" instead of "maybe". I think it's a bug... basically it warns with or without the attribute unless it's actually used.

Reply:
fe12ea2c948024693eb317635c091c060f1982e6 showed it when it wasn't used. This is on GCC 8.6 I think. When I have a moment to breathe I want to add some "oldest supported" compilers and dependencies to another CI toolchain so that we can be sure everything runs.